### PR TITLE
Add SSL support to Spring Config Server extension

### LINKED ISSUE #11293: Native SSL Support for Spring Cloud Config for Native Images
Quarkus has native SSL support with many if its different extensions.  However, not fir Spring Cloud Config where it is truly needed for native images.

I can think of two solutions.  1) allow a pluggable HTTP client that supports SSL out-of-the-box, or 2) Change to an HTTP client that already support it.

Thanks for your consideration.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #11293

### CODE REVIEW

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java [hunk 1]
```diff
@@ -58,13 +82,81 @@ private URI determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClie
         return new URI(url);
     }
 
+    // The SSL code is basically a copy of the code in the Consul extension
+    // Normally we would consider moving this code to once place, but as I want
```
Comment (gsmet) [inline extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java:86]:
```suggestion
    // Normally we would consider moving this code to one place, but as I want
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a small typo suggestion.

Wondering if we should backport it or not?

**Comment by geoand** [comment]
> Wondering if we should backport it or not?

We could consider it for 1.7.2

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by geoand** [comment]
I'll add the backport label, as this is safe for 1.7.2 and it is indeed pretty useful

**Review by gsmet — APPROVED** [review]

**Pull request merged**